libs/partners/elasticsearch/langchain_elasticsearch/vectorstores.py

```diff
@@ -389,6 +394,72 @@ def require_inference(self) -> bool:
         return False
 
 
+class BM25RetrievalStrategy(BaseRetrievalStrategy):
+    """Retrieval strategy using the native BM25 algorithm of Elasticsearch."""
+
+    def __init__(self, k1: float, b: float):
+        self.k1 = k1
+        self.b = b
+
+    def query(
+        self,
+        query_vector: Union[List[float], None],
+        query: Union[str, None],
+        k: int,
+        fetch_k: int,
+        vector_query_field: str,
+        text_field: str,
+        filter: List[dict],
+        similarity: Union[DistanceStrategy, None],
+    ) -> Dict:
+        return {
+            "query": {
+                "bool": {
+                    "must": [
+                        {
+                            "match": {
+                                text_field: {
+                                    "query": query,
+                                }
+                            },
+                        },
+                    ],
+                    "filter": filter,
+                },
+            },
+        }
+
+    def index(
+        self,
+        dims_length: Union[int, None],
+        vector_query_field: str,
+        text_field: str,
+        similarity: Union[DistanceStrategy, None],
+    ) -> Dict:
+        return {
+            "mappings": {
+                "properties": {
+                    text_field: {
+                        "type": "text",
+                        "similarity": "custom_bm25",
+                    },
+                },
+            },
+            "settings": {
+                "similarity": {
+                    "custom_bm25": {
+                        "type": "BM25",
+                        "k1": self.k1,
+                        "b": self.b,
```

Comment:
The `k1` and `b` settings should be given in the mapping only when custom values have been passed. If they are `None` then they should not be included, to allow Elasticsearch to use its own defaults.

Reply:
Thank you for pointing that out. I have addressed it in this commit: https://github.com/langchain-ai/langchain/pull/19314/commits/31f7909f906f8efdd1fefe42ec185b420e0d2191